allensdk/test/api/cloud_cache/test_manifest.py

```diff
@@ -0,0 +1,348 @@
+import pytest
+import json
+import io
+import pathlib
+from allensdk.internal.core.lims_utilities import safe_system_path
+from allensdk.api.cloud_cache.manifest import Manifest
+from allensdk.api.cloud_cache.file_attributes import CacheFileAttributes  # noqa: E501
+
+
+def test_constructor():
+    """
+    Make sure that the Manifest class __init__ runs and
+    raises an error if you give it an unexpected cache_dir
+    """
+    _ = Manifest('my/cache/dir')
+    _ = Manifest(pathlib.Path('my/other/cache/dir'))
+    with pytest.raises(ValueError) as context:
+        _ = Manifest(1234.2)
+    msg = "cache_dir must be either a str or a pathlib.Path; "
+    msg += "got <class 'float'>"
+    assert context.value.args[0] == msg
+
+
+def test_load(tmpdir):
+    """
+    Bare bones check to verify that Manifest.load can be run and that it
+    will raise the correct error when the JSONized manifest is not a dict
+    """
+
+    good_manifest = {}
+    good_manifest['dataset_version'] = 'A'
+    good_manifest['file_id_column'] = 'file_id'
+    metadata_files = {}
+    metadata_files['z.txt'] = []
+    metadata_files['x.txt'] = []
+    metadata_files['y.txt'] = []
+    good_manifest['metadata_files'] = metadata_files
+
+    stream = io.StringIO()
+    stream.write(json.dumps(good_manifest))
+    stream.seek(0)
+
+    mfest = Manifest(pathlib.Path(tmpdir) / 'my/cache/dir')
+    mfest.load(stream)
+    assert mfest.version == 'A'
+    assert mfest.metadata_file_names == ['x.txt', 'y.txt', 'z.txt']
+    assert mfest._cache_dir == pathlib.Path(str(tmpdir)+'/my/cache/dir')
+
+    del stream
+
+    # test that you can load a new manifest.json into the same Manifest
+    good_manifest = {}
+    good_manifest['dataset_version'] = 'B'
+    good_manifest['file_id_column'] = 'file_id'
+    metadata_files = {}
+    metadata_files['n.txt'] = []
+    metadata_files['k.txt'] = []
+    metadata_files['u.txt'] = []
+    good_manifest['metadata_files'] = metadata_files
+
+    stream = io.StringIO()
+    stream.write(json.dumps(good_manifest))
+    stream.seek(0)
+
+    mfest.load(stream)
+    assert mfest.version == 'B'
+    assert mfest.metadata_file_names == ['k.txt', 'n.txt', 'u.txt']
+
+    del stream
```

Comment:
should not these streams be `closed()`?
https://docs.python.org/3/library/io.html#io.StringIO

Reply:
You're right. I've wrapped them all in context managers so that they will be closed.